x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/edit_role_mapping_page.tsx

```diff
@@ -211,6 +214,49 @@ export class EditRoleMappingPage extends Component<Props, State> {
     );
   };
 
+  private checkEmptyAnyAllMappings = (obj: RoleMappingRule) => {
+    const arrToCheck: RoleMappingRule[] = [obj];
+
+    while (arrToCheck.length > 0) {
+      const currentObj = arrToCheck.pop();
+      if (typeof currentObj === 'object' && currentObj !== null) {
+        for (const key in currentObj) {
+          if (currentObj.hasOwnProperty(key)) {
+            const value: unknown = currentObj[key as keyof RoleMappingRule];
+            if ((key === 'any' || key === 'all') && Array.isArray(value)) {
```

Comment:
nit
I actually liked your example with `is`
Do you think it would be worth creating utility checks for those?
```typescript
const isRoleMappingAnyRule = (rule: RoleMappingRule): rule is RoleMappingAnyRule => {
  return rule.hasOwnProperty('any') && Array.isArray(rule.any);
};

const isRoleMappingAllRule = (rule: RoleMappingRule): rule is RoleMappingAllRule => {
  return rule.hasOwnProperty('all') && Array.isArray(rule.all);
}
```
So we could then use
```suggestion
            if (isRoleMappingAllRule(currentObj) || isRoleMappingAnyRule(currentObj)) {
```

Reply:
Done in https://github.com/elastic/kibana/pull/189340/commits/cb7f4f90837d3a9892726d1c2f3a968ddc7b9eee 

I made it a little more explicit to check individually for `any` or `all` to satisfy TS 😅

Reply:
Nice!